### PR TITLE
add throttling to file uploads

### LINKED ISSUE #18114: Add some throttling to the new /upload/ api
Follow up for #18023 

### Describe the problem and steps to reproduce it:
1. Send a few consecutive requests to the new `api/v5/addons/upload/` endpoint
2. Check the API response

### What happened?
Each request is successfully processed

### What did you expect to happen?
The API should be throttled after several consecutive requests

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #18114 - renames the existing add-on submission throttles to differentiate them better.  Most of the code is throttling tests that I duplicated from the signing api (because they're different, and one day the signing api and it's tests will be deleted)

### CODE REVIEW

File: src/olympia/files/tests/test_views.py [hunk 1]
```diff
@@ -180,10 +177,134 @@ def test_cannot_retrieve_other_uploads(self):
...
+            response = self._create_post(ip=ip)
+            assert response.status_code == 429, response.content
+
+            # 'Burst' throttling is 1 minute, so 61 seconds later we should be
+            # allowed again.
+            frozen_time.tick(delta=timedelta(seconds=61))
+            response = self._create_post(ip=ip)
+            assert response.status_code == 201, response.content
+
+    def test_throttling_verb_ip_hourly(self):
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_views.py:202]: Just wondering why this test case has the word "verb" in it?

File: src/olympia/files/tests/test_views.py [hunk 2]
```diff
@@ -180,10 +177,134 @@ def test_cannot_retrieve_other_uploads(self):
...
+                self._add_fake_throttling_action(
+                    view_class=FileUploadViewSet,
+                    url=self.list_url,
+                    user=user_factory(),
+                    remote_addr=ip,
+                )
+
+            # At this point we should be throttled since we're using the same
+            # IP. (we're still inside the frozen time context).
+            response = self._create_post(ip='63.245.208.194')
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_views.py:215]: Why not `ip=ip`?

File: src/olympia/files/tests/test_views.py [hunk 3]
```diff
@@ -180,10 +177,134 @@ def test_cannot_retrieve_other_uploads(self):
...
+            for _ in range(0, 6):
+                self._add_fake_throttling_action(
+                    view_class=FileUploadViewSet,
+                    url=self.list_url,
+                    user=self.user,
+                    remote_addr=get_random_ip(),
+                )
+
+            # At this point we should be throttled since we're using the same
+            # IP. (we're still inside the frozen time context).
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_views.py:241]: Should this comment say "using the same user", not "using the same IP"?

File: src/olympia/files/tests/test_views.py [hunk 4]
```diff
@@ -180,10 +177,134 @@ def test_cannot_retrieve_other_uploads(self):
...
+            for _ in range(0, 20):
+                self._add_fake_throttling_action(
+                    view_class=FileUploadViewSet,
+                    url=self.list_url,
+                    user=self.user,
+                    remote_addr=get_random_ip(),
+                )
+
+            # At this point we should be throttled since we're using the same
+            # IP. (we're still inside the frozen time context).
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_views.py:262]: Should this comment say "using the same user", not "using the same IP"?

File: src/olympia/files/tests/test_views.py [hunk 5]
```diff
@@ -180,10 +177,134 @@ def test_cannot_retrieve_other_uploads(self):
...
+            for _ in range(0, 48):
+                self._add_fake_throttling_action(
+                    view_class=FileUploadViewSet,
+                    url=self.list_url,
+                    user=self.user,
+                    remote_addr=get_random_ip(),
+                )
+
+            # At this point we should be throttled since we're using the same
+            # IP. (we're still inside the frozen time context).
```
Comment (bobsilverberg) [inline src/olympia/files/tests/test_views.py:288]: Should this comment say "using the same user", not "using the same IP"?

**Review by bobsilverberg — APPROVED** [review]
Looks good, just a few nits and a question.

r+wc

**Comment by eviljeff** [comment]
https://github.com/mozilla/addons-server/pull/18422/commits/b49136b1624a4b0af4a278cc0b150bbdfbf6d628 addresses the review comments

**Pull request merged**